### ISSUE TITLE
Which bin file for ESP32-C6 WROOM-N8

### ISSUE THREAD

**Issue by menaibluey** [rstrouse/ESPSomfy-RTS#381]
### Describe you new feature you'd like

What is the recommended firmware bin file to flash a ESP32-C6 version? Thanks.

**Comment by rstrouse** [comment]
I have not built firmware for the C6 yet.  Afik there is no Arduino core in release for it yet.

**Comment by menaibluey** [comment]
Ok thanks. Do you think that any of the other bin files will work. I don’t intend to use any of the zigbee or thread capabilities. Sent from my iPhoneOn 23 May 2024, at 09:35, rstrouse ***@***.***> wrote:﻿
I have not built firmware for the C6 yet.  Afik there is no Arduino core in release for it yet.

**Comment by rstrouse** [comment]
You can try the C3 firmware but I think the Wi-Fi libraries changed.

**Comment by bpat227** [comment]
I tried the C3 firmware but my C6 doesn't boot:
```
Invalid chip id. Expected 13 read 5. Bootloader for wrong chip?
```
If I try to compile it with Arduino 1.8.19 on my M1 Mac it fails with many errors, at least with 
```
In file included from SomfyController/Utils.cpp:3:
Utils.h:16:30: error: 'String' in namespace 'std' does not name a type
   16 |   template <typename T> std::String to_string(const T& t) {
      |                              ^~~~~~
```
Do you have any ideas?
I'm completly new to this environments.

**Comment by rstrouse** [comment]
It appears the Arduino Core v3.x was released last week.  One of the things that 3.x includes is support for the C6 generation of chips.  It will take a bit to get this transition over from v2.x since there are some breaking changes.

**Comment by rstrouse** [comment]
Unfortunately, the Arduino Core v3.x is 200k bigger than previous versions.  This will require a major code rework to which I have already started.  It will not fit on the ESP32 without removing features or repartitioning.  In the meantime this means that there is no support for the C6 chips yet.

The compile error you are getting is also because the std namespaces have been removed.  I suspect that this has something to do with trying to put the Core on a diet.
